SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/AlwaysInterestedTest/AlwaysInterestedTest.cpp

```diff
@@ -0,0 +1,335 @@
+	: Super()
+{
+	Author = "Mike";
+	Description = TEXT("Test Always Interested Functionality");
+}
+
+void AAlwaysInterestedTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ThisClass, ActorWithAlwaysInterestedProperty);
+	DOREPLIFETIME(ThisClass, InterestedInThisReplicatedActor);
+	DOREPLIFETIME(ThisClass, NotInterestedInThisReplicatedActor);
+	DOREPLIFETIME(ThisClass, OtherInterestedInThisReplicatedActor);
+}
+
+void AAlwaysInterestedTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	const float StepTimeLimit = 5.0f;
+
+	{ // Step 0 - Cache worker positions, and spawn actors on auth server
+		AddStep(TEXT("AlwaysInterested_SpawnActors"), FWorkerDefinition::AllServers, nullptr, [this]() {
+			ULayeredLBStrategy* RootStrategy = GetLoadBalancingStrategy();
+			UAbstractLBStrategy* DefaultStrategy = RootStrategy->GetLBStrategyForLayer(SpatialConstants::DefaultLayer);
+			UGridBasedLBStrategy* GridStrategy = Cast<UGridBasedLBStrategy>(DefaultStrategy);
+			AssertIsValid(GridStrategy, TEXT("Invalid LBS"));
+			const UGridBasedLBStrategy::LBStrategyRegions WorkerRegions = GridStrategy->GetLBStrategyRegions();
+			const VirtualWorkerId LocalWorker = GridStrategy->GetLocalVirtualWorkerId();
+
+			auto GetWorkerPosition = [&](bool bLocalWorker) -> FVector {
+				for (const auto& WorkerRegion : WorkerRegions)
+				{
+					if ((bLocalWorker && WorkerRegion.Key == LocalWorker) || (!bLocalWorker && WorkerRegion.Key != LocalWorker))
+					{
+						const FVector2D Centre = WorkerRegion.Value.GetCenter();
+						return FVector{ Centre.X, Centre.Y, 0.f };
+					}
+				}
+				return {};
+			};
+
+			LocalWorkerPosition = GetWorkerPosition(true);
+			OtherWorkerPosition = GetWorkerPosition(false);
+
+			if (HasAuthority())
+			{
+				ActorWithAlwaysInterestedProperty =
+					GetWorld()->SpawnActor<AAlwaysInterestedTestActor>(LocalWorkerPosition, FRotator::ZeroRotator, FActorSpawnParameters());
+
+				InterestedInThisReplicatedActor =
+					GetWorld()->SpawnActor<ASmallNCDActor>(LocalWorkerPosition, FRotator::ZeroRotator, FActorSpawnParameters());
+
+				NotInterestedInThisReplicatedActor =
+					GetWorld()->SpawnActor<ASmallNCDActor>(LocalWorkerPosition, FRotator::ZeroRotator, FActorSpawnParameters());
+
+				// This actor is used later as a replacement for InterestedInThisReplicatedActor, so isn't immediate added to
+				// AlwaysInterested
+				OtherInterestedInThisReplicatedActor =
+					GetWorld()->SpawnActor<ASmallNCDActor>(LocalWorkerPosition, FRotator::ZeroRotator, FActorSpawnParameters());
+
+				ActorWithAlwaysInterestedProperty->InterestedActors.Push(InterestedInThisReplicatedActor);
+
+				AController* PlayerController1 =
+					Cast<AController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+				if (!AssertTrue(IsValid(PlayerController1), TEXT("Failed to retrieve player controller 1")))
```

Comment:
Super nit: but I think I prefer "Should have spawned a PlayerController" "Should have authority over the player controller" instead of failed immediately in the message. That way when it fails, it says "Failed: should have auth over PC" or "Passed: should have auth over PC"

Reply:
but as I said, nit